identity/src/main/java/com/android/identity/internal/Util.java

```diff
@@ -245,7 +245,12 @@ DataItem cborDecode(@NonNull byte[] encodedBytes) {
     }
 
     public static boolean cborDecodeBoolean(@NonNull byte[] data) {
-        SimpleValue simple = (SimpleValue) cborDecode(data);
+        SimpleValue simple;
+        try {
+            simple = (SimpleValue) cborDecode(data);
+        } catch (ClassCastException e) {
+            throw new IllegalArgumentException("data given cannot be cast into a boolean");
+        }
         return simple.getSimpleValueType() == SimpleValueType.TRUE;
```

Comment:
I don't think we want to change the semantics of `Util.cborDecodeBoolean()` - lots of existing code depend on it. You should update the call-site instead.